fMRIVolume/scripts/MotionCorrection.sh

```diff
@@ -44,6 +59,30 @@ case "$MotionCorrectionType" in
     ;;
 esac
 
+
+# If registering across runs, perform nonlinear registration if requested.
+# (If using linear registration, don't need to do anything extra here, since
+# linear registration is handled implicitly via the motion correction).
+# Note that if registering across runs, the "$Scout" input to MotionCorrection will
+# be the *reference* scout image (by construction in GenericfMRIVolume).
+
+if [ "${fMRIReferenceReg}" == "nonlinear" ] ; then
+  verbose_echo " ... computing nonlinear transform to reference"
+  verbose_echo "     ... generating bold average"
+  # Generating a mean image to increase SNR to noise ratio when registering to scout.
+  ${FSLDIR}/bin/fslmaths ${WorkingDirectory}/${OutputfMRIBasename} -Tmean ${WorkingDirectory}/${OutputfMRIBasename}_avg
+
+  verbose_echo "     ... running fnirt: fnirt --in=${WorkingDirectory}/${OutputfMRIBasename}_avg --ref=${Scout} --iout=${WorkingDirectory}/${OutputfMRIBasename}_avg_nonlin --cout=${WorkingDirectory}/postmc2fmriref_warp"
+  ${FSLDIR}/bin/fnirt --in=${WorkingDirectory}/${OutputfMRIBasename}_avg --ref=${Scout} --iout=${WorkingDirectory}/${OutputfMRIBasename}_avg_nonlin --cout=${WorkingDirectory}/postmc2fmriref_warp
```

Comment:
If you want to echo the full command call, it would be cleaner, and less prone to error, if you set up the command just once.  e.g.,
```
cmd="${FSLDIR}/bin/fnirt --in=..."
verbose_echo "         ... running ${cmd}"
${cmd}
```

Reply:
ok.

Reply:
ok

Reply:
Please don't use that whitespace-unsafe construction, if you want to do this, use this array-based pattern:

```
cmd=(${FSLDIR}/bin/fnirt --in="...")
verbose_echo "         ... running fnirt: ${cmd[*]}"
"${cmd[@]}"
```
The `*` in the echo line is to prevent it from splitting the echo string into multiple arguments, the `@` in the actual command line inside `""` causes it to break array elements into arguments while being safe for whitespace.

Reply:
... and we could have just the uncomplicated initial version ... changed now. Again.

Reply:
The initial version wasn't whitespace safe, but could have been just by adding some quotes.  @mharms' version had the advantage of only defining the arguments once, so that editing it doesn't report the wrong command, but adding quotes would never make it whitespace-safe.  My version adds whitespace safety and removes the repetition, and making the initial version whitespace-safe would have needed all of those quotes anyway, and wouldn't resolve the repetition issue.  It is an improvement.

My comment was also intended for @mharms, so that he might avoid suggesting/using the unsafe pattern in the future.  No, we don't expect you to make these scripts completely whitespace-safe, but as long something is being edited, the new code shouldn't use bad patterns.